packages/screens/Launchpad/LaunchpadApply/LaunchpadCreate/components/LaunchpadStepper.tsx

```diff
@@ -0,0 +1,245 @@
+import { Dispatch, FC, SetStateAction, useRef } from "react";
+import { useFormContext } from "react-hook-form";
+import {
+  LayoutChangeEvent,
+  ScrollView,
+  TouchableOpacity,
+  useWindowDimensions,
+  View,
+} from "react-native";
+
+import ChevronRightSvg from "@/assets/icons/chevron-right.svg";
+import RejectSVG from "@/assets/icons/reject.svg";
+import { BrandText } from "@/components/BrandText";
+import { SVG } from "@/components/SVG";
+import { PrimaryBox } from "@/components/boxes/PrimaryBox";
+import { useIsMobile } from "@/hooks/useIsMobile";
+import {
+  neutral17,
+  neutral22,
+  neutral77,
+  primaryColor,
+  primaryTextColor,
+} from "@/utils/style/colors";
+import { fontMedium14 } from "@/utils/style/fonts";
+import { layout } from "@/utils/style/layout";
+import { CollectionFormValues } from "@/utils/types/launchpad";
+export type LaunchpadCreateStepKey = number;
+
+interface LaunchpadStepperProps {
+  selectedStepKey: LaunchpadCreateStepKey;
+  setSelectedStepKey: Dispatch<SetStateAction<LaunchpadCreateStepKey>>;
+}
+
+interface LaunchpadCreateStep {
+  key: LaunchpadCreateStepKey;
+  title: string;
+}
+
+const steps: LaunchpadCreateStep[] = [
+  {
+    key: 1,
+    title: "Basic",
+  },
+  {
+    key: 2,
+    title: "Details",
+  },
+  {
+    key: 3,
+    title: "Team & Investments",
+  },
+  {
+    key: 4,
+    title: "Additional",
+  },
+  {
+    key: 5,
+    title: "Minting",
+  },
+  {
+    key: 6,
+    title: "Assets & Metadata",
+  },
+];
+
+export const LaunchpadStepper: FC<LaunchpadStepperProps> = ({
+  selectedStepKey,
+  setSelectedStepKey,
+}) => {
+  const { width: windowWidth } = useWindowDimensions();
+  const scrollViewRef = useRef<ScrollView>(null);
+  const isMobile = useIsMobile();
+  const collectionForm = useFormContext<CollectionFormValues>();
+
+  const hasErrors = (stepKey: number) => {
```

Comment:
Is react hook form does not give any way to do this more clean ?
Or could we maybe do like a map[step]array of field key and loop through the fields corresponding to the key to avoid having too much repetition ?
Wdyt ?